### PR TITLE
Endpoing routing 3.0 registration

### PR DESCRIPTION
**Pull request by JamesNK**
Temp PR to review changes

### CODE REVIEW

**Comment by rynowak** [comment]
It says no merge, yet there is a merge button.

**Comment by JamesNK** [comment]
🆙 📅

File: samples/MvcSandbox/Controllers/HomeController.cs [hunk 1]
```diff
@@ -2,6 +2,7 @@
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using Microsoft.AspNetCore.Mvc;
+using MvcSandbox.AuthorizationMiddleware;
```
Comment (rynowak) [inline samples/MvcSandbox/Controllers/HomeController.cs]: Doesn't seem super needed

File: samples/MvcSandbox/HealthChecks/HealthChecksEndpointRouteBuilderExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    {
+        private static readonly Random _random = new Random();
+
+        public static IEndpointConventionBuilder MapHealthChecks(this IEndpointRouteBuilder builder, string pattern)
+        {
+            return builder.MapGet(
+                pattern,
+                async httpContext =>
+                {
+                    await httpContext.Response.WriteAsync(_random.Next() % 2 == 0 ? "Up!" : "Down!");
```
Comment (rynowak) [inline samples/MvcSandbox/HealthChecks/HealthChecksEndpointRouteBuilderExtensions.cs]: This is pretty close approximation of how health checks work.

File: samples/MvcSandbox/Startup.cs [hunk 3]
```diff
@@ -20,14 +27,46 @@ public void ConfigureServices(IServiceCollection services)
...
+                    displayName: "Home");
+
+                builder.MapMvcRoute(
                     name: "default",
                     template: "{controller=Home}/{action=Index}/{id?}");
+
+                builder.MapMvcControllers();
+                builder.MapRazorPages();
+
+                builder.MapHealthChecks("/healthz");
```
Comment (rynowak) [inline samples/MvcSandbox/Startup.cs]: I'm liking this real lots.

File: src/Microsoft.AspNetCore.Mvc.Core/Builder/MvcEndpointRouteBuilderExtensions.cs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    public static class MvcEndpointRouteBuilderExtensions
+    {
+        public static IEndpointConventionBuilder MapMvcControllers(
+            this IEndpointRouteBuilder routeBuilder)
+        {
+            return MapMvcControllers<ControllerBase>(routeBuilder);
+        }
+
+        public static IEndpointConventionBuilder MapMvcControllers<TController>(
+            this IEndpointRouteBuilder routeBuilder) where TController : ControllerBase
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Core/Builder/MvcEndpointRouteBuilderExtensions.cs]: This is a neat idea. Controllers don't actually require inheriting controllerbase though.

**Pull request merged**